lib/engine/game/g_1880/step/check_fi_connection.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class CheckFIConnection < Engine::Step::Base
+          ACTIONS = %w[destination_connection].freeze
+
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] if entity.minor?
```

Comment:
I'm confused. The check happens on the foreign investors turn. Won't this prevent foreign investors from doing the action?

Reply:
argh, i misread the rules, I thought it was done at the end of the corp turn. 
however:
"At the end of each turn of each Foreign Investor, it is checked if there is a joint route network (see box) with the associated com"